### PR TITLE
Tidy up IAM code

### PR DESCRIPTION
**Pull request by philandstuff**
This does quite a lot of stuff -- probably easier to read individual commit messages

### CODE REVIEW

File: create-code-deploy-application.sh [hunk 1]
```diff
@@ -2,18 +2,9 @@
...
-
-fetchInstanceProfile(){
-    CODE_DEPLOYER_PROFILE_ARN=$(aws iam get-instance-profile \
-                                --instance-profile-name $INSTANCE_PROFILE_NAME \
-                                --query InstanceProfile.Roles[0].Arn \
-                                --output text)
-
-    echo "$CODE_DEPLOYER_PROFILE_ARN"
-}
+CODE_DEPLOYER_ROLE_ARN=arn:aws:iam::022990953738:role/code-deploy-role
```
Comment (om-sharma) [inline create-code-deploy-application.sh]: can we fetch ARN by the role name instead of hardcoding it?

File: create-instance.sh [hunk 2]
```diff
@@ -54,6 +54,72 @@ set_up_security_group() {
 
 }
 
+create_instance_profile() {
```
Comment (om-sharma) [inline create-instance.sh]: it is good to have a separate script to create instance profile, we can call the script here to create profile.

File: create-code-deploy-application.sh (hunk 1 above)
Comment (philandstuff) [inline create-code-deploy-application.sh]: discussed IRL. the ARN is entirely based on the role name so this isn't a big deal.

File: create-instance.sh (hunk 2 above)
Comment (philandstuff) [inline create-instance.sh]: yup, will do

**Comment by philandstuff** [comment]
@om-sharma updated

**Pull request merged**